src/components/MultiButtonWithSubQuery.tsx

```diff
@@ -15,7 +15,21 @@ export const navigateWithSubQuery = ({
 }: {
   navigation: StackNavigationProp<any, string>;
   title?: string;
-  params?: { routeName: string; webUrl: string } | string;
+  params?:
+    | {
```

Comment:
What does `| {` mean here?

Reply:
Isn't `{ routeName: string; webUrl: string }` missing in first place for cases, where not "multi button"?

Reply:
I don't know why you added the `| {` code. I think it was added during automatic formatting

---

If it is not multi button, the params object is not sent to this function as a prop.

```
return (
    <>
      {!!subQuery && !!subQuery.routeName && (!!subQuery.webUrl || subQuery.params) && (
        <Button
          title={subQuery.buttonTitle || `${title} öffnen`}
          onPress={() => navigateWithSubQuery({ navigation, subQuery, rootRouteName, title })}
        />
      )}

     ...
    </>
  );
```

Reply:
i cannot understand how "If it is not multi button, the params object is not sent to this function as a prop." relates to the code you have sent? can you elaborate a bit more please. the `params` are part of `subQuery` which is passed to `navigateWithSubQuery`.

Reply:
The `params` prop specified in the question is only valid for multi buttons. We do not send the `params` prop to the `navigateWithSubQuery` function with the first button on the screen. instead we create a variable called `subParams` in the `navigateWithSubQuery` function and integrate `subQuery.params` into the navigation structure. That's why I thought the params prop is not necessary for the first button, so I sent the code required for the first button.

Or I have misunderstood the question you wanted to ask. 🤷‍♂️
```
export const navigateWithSubQuery = ({
...
const subParams = { ...(subQuery.params ?? {}) };

  // if there is no `params`, use the main `subQuery` values for `routeName` and a `webUrl` or
  // `params` if the params contain a webUrl as well, the webUrl property of the subQuery
  // will be ignored
  return navigation.navigate({
    name: subQuery.routeName,
    params: {
      ...subParams,
      rootRouteName,
      title,
      webUrl: subQuery.webUrl
    }
  });

...
return (
    <>
      {!!subQuery && !!subQuery.routeName && (!!subQuery.webUrl || subQuery.params) && (
        <Button
          title={subQuery.buttonTitle || `${title} öffnen`}
          onPress={() => navigateWithSubQuery({ navigation, subQuery, rootRouteName, title })}
        />
      )}

      {subQuery?.buttons?.map((button, index) => (
        <Button
          key={`${index}-${button.webUrl}`}
          title={button.buttonTitle || `${title} öffnen`}
          onPress={() =>
            navigateWithSubQuery({
              navigation,
              params: button,
              rootRouteName,
              subQuery,
              title
            })
          }
        />
      ))}
    </>
  );
```

Reply:
got it